### ISSUE TITLE
Swipe duration not translated correctly in Appium 1.6.0

### ISSUE THREAD

**Issue by VivaLosDoyers** [appium/appium#7047]
## The problem

Attempting to swipe in Appium 1.6.0 is giving me an error that duration values must be > 0.5 and < 60, even though I am supplying a valid value. The appium log shows that my supplied value of 50 is being changed to 0.025, which is < 0.5. But why is it suddenly doing this?

This is with Appium 1.6.0, using UIAutomation (not XCUITest, which apparently doesn't even have swipe implemented yet...)

For example:
`Appium::TouchAction.new.swipe(start_x: 200, start_y: 200, end_x: 100, end_y:100, duration:50).perform`

Shows the following in the Appium log:

```
[HTTP] --> POST /wd/hub/session/29ffaad2-ae08-4d02-9764-04a56d8ccc09/touch/perform {"actions":[{"action":"press","options":{"x":200,"y":200}},{"action":"wait","options":{"ms":25}},{"action":"moveTo","options":{"x":100,"y":100}},{"action":"release"}]}
[MJSONWP] Calling AppiumDriver.performTouch() with args: [[{"action":"press","option...
[debug] [iOS] Executing iOS command 'performTouch'
[debug] [UIAuto] Sending command to instruments: au.dragApp(200, 200, 300, 300, 0.025)
[debug] [Instruments] [INST] 2016-10-18 19:09:29 +0000 Debug: Got new command 13 from instruments: au.dragApp(200, 200, 300, 300, 0.025)
[debug] [Instruments] [INST] 2016-10-18 19:09:29 +0000 Debug: evaluating au.dragApp(200, 200, 300, 300, 0.025)
[debug] [Instruments] [INST] 2016-10-18 19:09:29 +0000 Debug: target.dragFromToForDuration({x:"200", y:"200"}, {x:"300", y:"300"}, "0.025")
[debug] [Instruments] [INST] 2016-10-18 19:09:29 +0000 Debug: duration value must be greater than or equal to 0.5 or less than 60
[debug] [Instruments] [INST] 2016-10-18 19:09:29 +0000 Error: VerboseError: duration value must be greater than or equal to 0.5 or less than 60
```
## Environment
- Appium version (or git revision) that exhibits the issue: 1.6.0
- Last Appium version that did not exhibit the issue (if applicable): 1.5.3
- Desktop OS/version used to run Appium: OS X 10.12
- Node.js version (unless using Appium.app|exe): 6.7.0
- Mobile platform/version under test: iOS 9.3, Xcode 7.3.1
- Real device or emulator/simulator: simulator
- Appium CLI or Appium.app|exe: CLI

**Comment by imurchie** [comment]
What language/client are you using?

**Comment by VivaLosDoyers** [comment]
Ruby. I know ruby_lib is way out of date. Is that what could be causing this?

**Comment by imurchie** [comment]
I don't think so. I'm just confused by why when you send in `50` for duration it becomes `25` when sent to the server (`{"action":"wait","options":{"ms":25}}`).

**Comment by imurchie** [comment]
It should just be passed in: https://github.com/appium/ruby_lib/blob/11b80e398e98fbc71e580f659764ba54f87da4f3/lib/appium_lib/device/touch_actions.rb#L148

**Comment by VivaLosDoyers** [comment]
Let me try again. Maybe I grabbed the wrong part of the log.

**Comment by VivaLosDoyers** [comment]
If I pass duration: 500:

```
[HTTP] --> POST /wd/hub/session/87f65ca3-6861-4258-814c-a5812622c2aa/touch/perform {"actions":[{"action":"press","options":{"x":200,"y":200}},{"action":"wait","options":{"ms":500}},{"action":"moveTo","options":{"x":300,"y":300}},{"action":"release"}]}
[MJSONWP] Calling AppiumDriver.performTouch() with args: [[{"action":"press","option...
[debug] [iOS] Executing iOS command 'performTouch'
[debug] [UIAuto] Sending command to instruments: au.dragApp(200, 200, 500, 500, 0.5)
[debug] [Instruments] [INST] 2016-10-18 19:37:31 +0000 Debug: Got new command 6 from instruments: au.dragApp(200, 200, 500, 500, 0.5)
[debug] [Instruments] [INST] 2016-10-18 19:37:31 +0000 Debug: evaluating au.dragApp(200, 200, 500, 500, 0.5)
[debug] [Instruments] [INST] 2016-10-18 19:37:31 +0000 Debug: target.dragFromToForDuration({x:"200", y:"200"}, {x:"500", y:"500"}, "0.5")
[debug] [Instruments] [INST] 2016-10-18 19:37:32 +0000 Debug: evaluation finished
[debug] [Instruments] [INST] 2016-10-18 19:37:32 +0000 Debug: responding with:
[debug] [Instruments] [INST] 2016-10-18 19:37:32 +0000 Debug: Running system command #7: /usr/local/Cellar/node/6.7.0/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js /var/folders/gb/lyxrv3116vlfkx68nt22s7rm0000gq/T/instruments_sock 2,{"status":0,"value":""}...
[debug] [UIAuto] Socket data received (25 bytes)
[debug] [UIAuto] Got result from instruments: {"status":0,"value":""}
[MJSONWP] Responding to client with driver.performTouch() result: ""
[HTTP] <-- POST /wd/hub/session/87f65ca3-6861-4258-814c-a5812622c2aa/touch/perform 200 1974 ms - 74 

```

**Comment by VivaLosDoyers** [comment]
It looks like I pass it ms, it gets translated into seconds, and then if I pass an invalid value, it reports valid values in the range of seconds instead of ms.

**Comment by imurchie** [comment]
Yes, the error message printed out is from Apple. The value should be in milliseconds, which is translated and passed to the underlying system..

**Comment by mobileGadget** [comment]
I have the same issue as well, albeit using the appium java-client v4.1.2, which is the latest available.
The inability to swipe is making a huge impact to my current test scripts.

**Comment by mobileGadget** [comment]
I have figured it out a way to make the swipe work now by increasing the duration value in appiumdriver.swipe(startx, starty, endx, endy, duration); from 100 to 6000.  I think that translate into 60ms which is a value between 0.5 and 60

**Comment by dpgraham** [comment]
Closing this. It looks the client is behaving as expected the logging is slightly confusing though, but we no longer support Instruments anyway.
